shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/AuthenticationService.kt

```diff
@@ -0,0 +1,52 @@
+package co.touchlab.droidcon.domain.service
+
+import co.touchlab.droidcon.UserContext
+import co.touchlab.droidcon.UserData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+abstract class AuthenticationService(isSignedIn: Boolean) : KoinComponent {
```

Comment:
Could it be possible to fit this a little better into the exiting architecture? If we remove the googleSignIn function from this class then I think we could make this class an interface with a default implementation in the `impl` folder like the other services here.

googleSignIn could then be it's own service interface, but it's implementation would live in the app layer (following the example of AnalyticsService)

Reply:
Ok, I was able to separate into an `AuthenticationService` and a `GoogleSignInService`, hopefully this matches the architecture a bit more

Reply:
looks great!